compiler/rustc_builtin_macros/src/deriving/generic/mod.rs

```diff
@@ -1532,11 +1579,76 @@ impl<'a> TraitDef<'a> {
                 let mut exprs: Vec<_> = mk_exprs(i, struct_field, sp);
                 let self_expr = exprs.remove(0);
                 let other_selflike_exprs = exprs;
+                let mut skipped_derives = SkippedDerives::None;
```

Comment:
Might want to move a lot of this code into a separate method to avoid rightward drift.